service/controller/app/resource/chartoperator/create_test.go

```diff
@@ -0,0 +1,400 @@
+package chartoperator
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/giantswarm/apiextensions-application/api/v1alpha1"
+	"github.com/giantswarm/app/v6/pkg/key"
+	"github.com/giantswarm/app/v6/pkg/values"
+	"github.com/giantswarm/helmclient/v4/pkg/helmclient"
+	"github.com/giantswarm/helmclient/v4/pkg/helmclienttest"
+	"github.com/giantswarm/k8sclient/v6/pkg/k8sclienttest"
+	"github.com/giantswarm/k8smetadata/pkg/label"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"github.com/spf13/afero"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgofake "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/giantswarm/app-operator/v5/service/controller/app/controllercontext"
+)
+
+func Test_Resource_EnsureCreated(t *testing.T) {
```

Comment:
The two `Test_` functions are now redundant, sort of, though I believe we could keep them anyway in case we extend test cases for the `EnsureCreated`.

Reply:
I think just having a UT for `triggerReconciliation` is enough.

The tests for client-go become verbose quickly and often an integration test is better. The chart-operator bootstrapping is tested by the WorkloadClusterTest so I don't think we need coverage for `EnsureCreated`.

https://github.com/giantswarm/app-operator/blob/c72fc6ad2601b0908cd0ae294fd3166082c0d6f0/integration/test/app/workload/workload_cluster_test.go#L29-L31

Reply:
Thanks @rossf7 , makes sense to me, I'm removing this